File: 566 - Reshape the Matrix.js
/**
 * @param {number[][]} nums
 * @param {number} r
 * @param {number} c
 * @return {number[][]}
 */
var matrixReshape = function (nums, r, c) {
    let col = nums[0].length;
    if (nums.length * col != r * c) {
        return nums;
    }
    let res = new Array();
    for (let i = 0; i < r; i++) {
        res[i] = new Array();
    }
    let count = 0;
    while (count < r * c) {
        res[Math.floor(count / c)][count % c] = nums[Math.floor(count / col)][count % col]
        count++;
    }
    return res;

};

matrixReshape([[1, 2], [3, 4]], 1, 4);

File: 566 - Reshape the Matrix.java
class Solution {
    public int[][] matrixReshape(int[][] nums, int r, int c) {
        int row = nums.length, col = nums[0].length;
        if (row * col != r * c) {
            return nums;
        }
        int[][] res = new int[r][c];
        int count = 0;
        while (count < r * c) {
            res[count / c][count % c] = nums[count / col][count % col];
            count++;
        }
        return res;
    }
}